### PR TITLE
Update API docs and structure.

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR brings the API section of the docs in line with the refactor work.  To-dos remain:
- mesh @peterdsharpe 
- diffusion @CharlelieLrt 
- datapipes @coreyjadams 
- nn @loliverhennigh 

We also need to check models are all listed in the API section, I think we are missing some.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: docs/api/physicsnemo.core.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Utilities for handling file operations, caching, and data management across different storage systems.
+These utilities abstract away the complexity of dealing with different filesystem types and provide
+consistent interfaces for data access.
+
+.. automodule:: physicsnemo.utils.filesystem
+    :members:
+    :show-inheritance:
+
+.. _diffusion_utils:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/physicsnemo.core.rst:29]: This will get fixed with the diffusion PR from @CharlelieLrt

File: docs/api/models/operators.rst [hunk 2]
```diff
@@ -5,3 +5,13 @@ Operator Models
...
+
+Domino utils
+------------
+
+Utilities for working with the Domino model, including data processing and grid construction.
+These utilities are essential for implementing and training Domino-based models.
+
+.. automodule:: physicsnemo.utils.domino.utils
+    :members:
+    :show-inheritance:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/models/operators.rst]: This is fixed now

File: docs/api/physicsnemo.diffusion.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Tools for working with diffusion models and other generative approaches,
+including deterministic and stochastic sampling utilities.
+
+.. automodule:: physicsnemo.diffusion.samplers.deterministic_sampler
+    :members:
+    :show-inheritance:
+
+.. automodule:: physicsnemo.diffusion.samplers.stochastic_sampler
+    :members:
+    :show-inheritance:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/physicsnemo.diffusion.rst:32]: This will get fixed in PR from @CharlelieLrt

File: docs/api/models/weather.rst [hunk 4]
```diff
@@ -15,6 +15,31 @@ Weather / Climate Models
...
+    :members:
+    :show-inheritance:
+
+.. automodule:: physicsnemo.graphcast.utils.graph_utils
+    :members:
+    :show-inheritance:
+
+.. automodule:: physicsnemo.graphcast.utils.loss
+    :members:
+    :show-inheritance:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/models/weather.rst]: This is fixed now.

File: docs/api/physicsnemo.nn.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    :show-inheritance:
+
+Neighbor utils
+--------------
+
+Utilities for optimized layers for neighbor searches.
+
+.. automodule:: physicsnemo.nn.neighbors
+    :members:
+    :show-inheritance:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/physicsnemo.nn.rst]: Updated this and removed the repetition from physicsnemo.nn.functionals.rst

**Review by peterdsharpe — APPROVED** [review]
Reviewing only the changes to the Mesh API docs - this migration looks good to me! Thanks!

**Review by loliverhennigh — APPROVED** [review]
Checked the image moves and it looks good to me. I read through the other doc changes. They make sense to me and happy to approve although I know others might know a bit better about the distributed stuff and diffusion.

File: docs/api/models/diffusion.rst [hunk 6]
```diff
@@ -418,3 +418,12 @@ implement a regression model.
     :show-inheritance:
     :members:
     :exclude-members: forward
+
+CorrDiff utils
```
Comment (megnvidia) [inline docs/api/models/diffusion.rst]:
```suggestion
CorrDiff Utils
```

File: docs/api/models/operators.rst [hunk 7]
```diff
@@ -5,3 +5,13 @@ Operator Models
     :show-inheritance:
     :members:
     :exclude-members: forward
+
+Domino utils
```
Comment (megnvidia) [inline docs/api/models/operators.rst]:
```suggestion
Domino Utils
```

File: docs/api/models/weather.rst [hunk 8]
```diff
@@ -15,6 +15,22 @@ Weather / Climate Models
 
 .. autofunction:: physicsnemo.models.graphcast.graph_cast_net.get_lat_lon_partition_separators
 
+GraphCast utils
```
Comment (megnvidia) [inline docs/api/models/weather.rst]:
```suggestion
GraphCast Utils
```

File: docs/api/physicsnemo.core.rst [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+of `physicsnemo.Module` and metadata, and the model registry.
+
+PhysicsNemo ``Module``
+----------------------
+
+The :class:`Module` provides a base class for all user facing models in PhysicsNeMo.
+It provides a unified interface for the registry, checkpointing, and optimization.
+Full API information can be found in the :doc:`PhysicsNeMo Modules <models/modules>`.
+
+Filesystem utils
```
Comment (megnvidia) [inline docs/api/physicsnemo.core.rst]:
```suggestion
Filesystem Utils
```

File: docs/api/physicsnemo.diffusion.rst [hunk 10]
```diff
@@ -0,0 +1,36 @@
+PhysicsNeMo Diffusion
+=====================
+
+Patching utils
```
Comment (megnvidia) [inline docs/api/physicsnemo.diffusion.rst]:
```suggestion
Patching Utils
```

File: docs/api/physicsnemo.diffusion.rst [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+smaller patches that are extracted from each sample in the original batch of larger
+images. Diffusion models can then process these patches independently. These
+utilities also support fusing operations to reconstruct the entire predicted
+image from the individual predicted patches.
+
+.. automodule:: physicsnemo.diffusion.multi_diffusion
+    :members:
+    :show-inheritance:
+
+Diffusion utils
```
Comment (megnvidia) [inline docs/api/physicsnemo.diffusion.rst]:
```suggestion
Diffusion Utils
```

File: docs/api/physicsnemo.domain_parallel.rst [hunk 12]
```diff
@@ -0,0 +1,29 @@
+
+PhysicsNeMo domain_parallel
```
Comment (megnvidia) [inline docs/api/physicsnemo.domain_parallel.rst]: I am having a serious debate with myself here.  I am 60% in favor of changing this to be in keeping with style guide suggestion to have code things in code font. But I don't think that pattern will predominantly match what you have in the rest of the docs. So I give you the freedom to overrule. 

PhysicsNeMo ``domain_parallel``

File: docs/api/physicsnemo.nn.rst
Comment (megnvidia) [inline docs/api/physicsnemo.nn.rst:1]: I like this description. Well done. Elegant

File: docs/api/physicsnemo.utils.rst [hunk 13]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
     :show-inheritance:
 
 
-GraphCast Utils
----------------
-
-A collection of utilities specifically designed for working with the GraphCast model, including data processing,
-graph construction, and specialized loss functions. These utilities are essential for implementing and
-training GraphCast-based weather prediction models.
+Weather / Climate utils
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
Weather and Climate Utils
```

File: docs/api/physicsnemo.utils.rst [hunk 14]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+------------------------
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
--------------------------
```

File: docs/api/physicsnemo.utils.rst [hunk 15]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+                ... # validation code
+                logger.log_minibatch({"val_loss": validation_loss})
+        
+        learning_rate = ... # get the learning rate at the end of the epoch from the optimizer
+        logger.log_epoch({"learning_rate": learning_rate}) # log the learning rate at the end of the epoch
+
+This example shows how to use the LaunchLogger to log metrics during training and
+validation. The LaunchLogger is initialized with the MLflow backend, and the logger
+is created for each epoch, a separate logger is created for training and validation.
+We can use the `.log_minibatch` method to log metrics during training and validation.
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
You can use the `.log_minibatch` method to log metrics during training and validation.
```

File: docs/api/physicsnemo.utils.rst [hunk 16]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+We can use the `.log_epoch` method to log the learning rate at the end of the epoch.
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
You can use the `.log_epoch` method to log the learning rate at the end of the epoch.
```

File: docs/api/physicsnemo.utils.rst [hunk 17]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+
+For a more detailed example, please refer to the `Logging and Checkpointing recipe <../../user-guide/simple_logging_and_checkpointing.html>`_ .
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
For a more detailed example, refer to the `Logging and Checkpointing recipe <../../user-guide/simple_logging_and_checkpointing.html>`_ .
```

File: docs/api/physicsnemo.utils.rst [hunk 18]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+    # Create logger with MLflow backend
+    logger = LaunchLogger.initialize(use_mlflow=True)
 
-.. automodule:: physicsnemo.diffusion.samplers
+Weights and Biases Logger
+-------------------------
+
+Integration with Weights & Biases (W&B) for experiment tracking and visualization. This utility provides rich
+visualization capabilities and easy experiment comparison, making it ideal for projects that require detailed
+analysis of training runs and model performance. Users should initialize the W&B backend before using the LaunchLogger.
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
analysis of training runs and model performance. You must initialize the W&B backend before using the LaunchLogger.
```

File: docs/api/physicsnemo.utils.rst [hunk 19]
```diff
@@ -23,115 +23,185 @@ These are particularly useful when implementing custom training loops or special
...
+    # Initialize W&B
+    initialize_wandb(
+        project="physics_simulation",
+        entity="my_team"
+    )
+    
+    # Create logger with W&B backend
+    logger = LaunchLogger.initialize(use_wandb=True)
+
+Logging utils
```
Comment (megnvidia) [inline docs/api/physicsnemo.utils.rst]:
```suggestion
Logging Utils
```

**Review by megnvidia — CHANGES_REQUESTED** [review]
